### PR TITLE
Adding ptmcmcsampler recipe

### PR DESCRIPTION
**Pull request by jellis18**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jellis18** [comment]
@conda-forge-admin, please ping team

File: recipes/ptmcmcsampler/meta.yaml
Comment (dopplershift) [inline recipes/ptmcmcsampler/meta.yaml:26]: I see mpi4py is an optional/extras dependency upstream. Does *every* conda user of this package need to install mpi?

File: recipes/ptmcmcsampler/meta.yaml
Comment (dopplershift) [inline recipes/ptmcmcsampler/meta.yaml:40]: Would be good to add a comment stating that the local copy of the license (in conda-forge) can be removed on the next release.

Comment (jellis18) [reply recipes/ptmcmcsampler/meta.yaml:40]: Should this comment go in `meta.yaml`?

File: recipes/ptmcmcsampler/meta.yaml
Comment (jellis18) [reply recipes/ptmcmcsampler/meta.yaml:26]: I'm pretty new to conda, can we do optional requirements like in with pip? If not, what is the recommended way to do this?

File: recipes/ptmcmcsampler/meta.yaml
Comment (dopplershift) [reply recipes/ptmcmcsampler/meta.yaml:40]: Yeah, I think that's fine. Otherwise, someone has to remember to open an issue when the recipe gets merged or to just remember to do it the next time the version bumps.

File: recipes/ptmcmcsampler/meta.yaml
Comment (dopplershift) [reply recipes/ptmcmcsampler/meta.yaml:26]: There's not great support for optional dependencies. One way is to just make it just generally required, like you've done. In that case it's always available, but *everyone*, even non-MPI users will end up with MPI installed.

The other option is to leave it off, and rely on users of MPI already having `mpi4py` installed or knowing enough to install it manually. Obviously, that has discoverability issues. So I leave it to you as the maintainer and package contributor what you think is best for the users of `ptmcmcsampler` or even just your own needs.

File: recipes/ptmcmcsampler/meta.yaml
Comment (jellis18) [reply recipes/ptmcmcsampler/meta.yaml:41]: Comment added.

File: recipes/ptmcmcsampler/meta.yaml
Comment (jellis18) [reply recipes/ptmcmcsampler/meta.yaml]: I've removed it. I'll add some instructions on the github page about installing this dependency if needed on conda.

**Comment by jellis18** [comment]
@dopplershift I think we are good to do now.

**Review by dopplershift — APPROVED** [review]

**Pull request merged**
